Remotion/Web/Development.WebTesting/ScreenshotCreation/BrowserContentLocators/EdgeBrowserContentLocator.cs

```diff
@@ -0,0 +1,112 @@
+                  new OrCondition (
+                      new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Window),
+                      new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Pane)),
+                  new PropertyCondition (AutomationElement.ClassNameProperty, "Chrome_WidgetWin_1")))
+          .Cast<AutomationElement>()
+          .ToArray();
+
+      if (windows.Length == 1)
+        return ResolveBoundsFromWindow (windows[0]);
+
+      if (windows.Length == 0)
+        throw new InvalidOperationException ("Could not find an Edge browser window in order to resolve the bounds of the content area.");
+
+      // If the result are ambiguous we try to find the browser by changing the window title
+      var automationElement = ResolveByChangingWindowTitle (driver, windows);
+
+      return ResolveBoundsFromWindow (automationElement);
+    }
+
+    [CanBeNull]
+    private AutomationElement ResolveByChangingWindowTitle (IWebDriver driver, IReadOnlyCollection<AutomationElement> windows)
+    {
+      var id = Guid.NewGuid().ToString();
+
+      var executor = (IJavaScriptExecutor) driver;
+      var previousTitle = JavaScriptExecutor.ExecuteStatement<string> (executor, c_setWindowTitle, id);
+
+      var result = RetryUntil (
+          () => windows.SingleOrDefault (w => w.Current.Name.StartsWith (id)),
+          null,
+          3,
+          TimeSpan.FromMilliseconds (100));
+
+      JavaScriptExecutor.ExecuteStatement<string> (executor, c_setWindowTitle, previousTitle);
+
+      if (result == null)
+        throw new InvalidOperationException ("Could not find a matching Edge window by changing its window title.");
+
+      return result;
+    }
+
+    private Rectangle ResolveBoundsFromWindow (AutomationElement window)
+    {
+      var element = RetryUntil (
+          () => window.FindFirst (
+              TreeScope.Subtree,
+              new AndCondition (
+                  new PropertyCondition (AutomationElement.ControlTypeProperty, ControlType.Document),
+                  new PropertyCondition (AutomationElement.FrameworkIdProperty, "Chrome"))),
+          null,
+          5,
+          TimeSpan.Zero);
+
+      if (element == null)
+        throw new InvalidOperationException ("Could not find the content window of the found Edge browser window.");
+
+      var rawBounds = RetryUntil (() => element.Current.BoundingRectangle, Rect.Empty, 3, TimeSpan.FromMilliseconds (100));
+
+      if (rawBounds == Rect.Empty)
+        throw new InvalidOperationException ("Could not resolve the bounds of the Edge browser window.");
+
+      return new Rectangle (
+          (int) Math.Round (rawBounds.X),
+          (int) Math.Round (rawBounds.Y),
+          (int) Math.Round (rawBounds.Width),
+          (int) Math.Round (rawBounds.Height));
+    }
+
+    private TResult RetryUntil<TResult> (Func<TResult> action, TResult defaultValue, int retries, TimeSpan interval)

```

Comment:
Rename parameter "action" to "function" etc. ?

Reply:
Also, what do you think about changing the name "defaultValue" to something like "valueToCompareTo" etc (to express that it's the value that actually triggers the retry if the result is that value)

Reply:
The `defaultValue` is the value that the `Func` result is compared with, and the value that is returned if the timeout is exceeded. Imo `defaultValue` expresses this good enough, I'll leave as-is until the lead review.